code/modules/projectiles/guns/misc/syringe_gun.dm

```diff
@@ -117,12 +117,20 @@
 	chambered = new /obj/item/ammo_casing/syringegun/dart(src)
 
 /obj/item/gun/syringe/dart/attackby(obj/item/A, mob/user, params, show_msg = TRUE)
-	if(istype(A, /obj/item/reagent_containers/syringe/dart))
+	if(istype(A, /obj/item/reagent_containers/syringe/dart) || (istype(A, /obj/item/reagent_containers/syringe) && obj_flags & EMAGGED))
 		..()
+	else if(istype(A, /obj/item/card/emag))
+		emag_act(user)
+		return TRUE
 	else
 		to_chat(user, "<span class='notice'>You can't put the [A] into \the [src]!</span>")
 		return FALSE
 
+/obj/item/gun/syringe/dart/emag_act(mob/living/user)
+	to_chat(user, "<span class='warning'>\the [src] will now accept any type of syringe.</span>")
+	playsound(user.loc, "sparks", 50, 1)
+	obj_flags |= EMAGGED
```

Comment:
It's supposed to return TRUE or FALSE so we know when to detract emag charges or not.
```suggestion
	. = ..()
	if(obj_flags & EMAGGED)
		return
	to_chat(user, "<span class='warning'>\the [src] will now accept any type of syringe.</span>")
	playsound(user.loc, "sparks", 50, 1)
	obj_flags |= EMAGGED
	return TRUE
```